lib/everypolitician/pull_request/report/memberships.rb

```diff
@@ -13,16 +13,15 @@ def removed
         private
 
         def before_h
-          @before_h ||= before.memberships.map { |m| [document_without_sources(m.document), m] }.to_h
+          @before_h ||= before.memberships.map { |m| [filtered_document(m.document), m] }.to_h
         end
 
         def after_h
-          @after_h ||= after.memberships.map { |m| [document_without_sources(m.document), m] }.to_h
+          @after_h ||= after.memberships.map { |m| [filtered_document(m.document), m] }.to_h
         end
 
-        def document_without_sources(doc)
-          doc.delete(:sources)
-          doc
+        def filtered_document(doc, ignore: %i(sources))
```

Comment:
This method has a little bit too much feature envy for my liking, and at a slightly odd level of abstraction too (i.e. not even on Membership, but on Document). I fear it's largely premature abstraction, but in a subtly wrong way that's going to bite us later, though I'm afraid that's largely just based on gut reaction and intuition, rather than being able to convincingly argue for it.

The over-engineered solution here would probably be to have our own context-specific wrapper or subclass of Membership, but as we've already simply extended Hash, I think I'd just inline both of these and drop `filtered_document` entirely. We end up with a little bit of duplication, but on methods that are already essentially twins anyway, and when we get to the point where that duplication is too much I suspect we'll want to tease them apart at a slightly different level anyway. 

I'll add that as a new commit to see what you think.